model/src/application/interaction/mod.rs

```diff
@@ -162,18 +143,19 @@ impl<'de> Visitor<'de> for InteractionVisitor {
 
     #[allow(clippy::too_many_lines)]
     fn visit_map<V: MapAccess<'de>>(self, mut map: V) -> Result<Self::Value, V::Error> {
+        let mut app_permissions: Option<Permissions> = None;
         let mut application_id: Option<Id<ApplicationMarker>> = None;
         let mut channel_id: Option<Id<ChannelMarker>> = None;
         let mut data: Option<Value> = None;
```

Comment:
I think this can be replaced
```suggestion
        let mut data: Option<InteractionData> = None;
```

Reply:
It would rely on the `#[serde(untagged)]` which I think is much slower since it try to deserialize each variant. The current implementation choose the right variant according to the interaction `type`.